src/main/java/com/targetindia/DigitizeTimeTable/controller/InstructorController.java

```diff
@@ -0,0 +1,107 @@
+//package com.targetindia.DigitizeTimeTable.controller;
+//
+//import com.targetindia.DigitizeTimeTable.service.AdminService;
+//import com.targetindia.DigitizeTimeTable.service.InstructorService;
+//import org.springframework.beans.factory.annotation.Autowired;
+//import org.springframework.web.bind.annotation.GetMapping;
+//import org.springframework.web.bind.annotation.PathVariable;
+//import org.springframework.web.bind.annotation.RequestMapping;
+//import org.springframework.web.bind.annotation.RestController;
+//
+//import java.util.ArrayList;
+//import java.util.HashMap;
+//import java.util.Optional;
+//
+//@RestController
+//@RequestMapping("instructor")
+//public class InstructorController {
+//    @Autowired
+//    InstructorService service;
+//
+//    @GetMapping("GET/time_table/instructors_id/{id}/day/{day}")
+//    public HashMap<String, ArrayList<String>> getInstructorDayTimeTable(@PathVariable int id, @PathVariable String day){
+//        return service.getInstructorDayTimeTable(id, day);
+//    }
+//    @GetMapping("GET/time_table/{id}")
+//    public HashMap<String,HashMap<String, ArrayList<String>>> getInstructorWeeklyTimeTable(@PathVariable int id){
+//        return service.getInstructorWeeklyTimeTable(id);
+//    }
+//}
+package com.targetindia.DigitizeTimeTable.controller;
+
+        import com.targetindia.DigitizeTimeTable.service.AdminService;
+        import com.targetindia.DigitizeTimeTable.service.InstructorService;
+        import org.springframework.beans.factory.annotation.Autowired;
+        import org.springframework.http.HttpStatusCode;
+        import org.springframework.http.ResponseEntity;
+        import org.springframework.web.bind.annotation.GetMapping;
+        import org.springframework.web.bind.annotation.PathVariable;
+        import org.springframework.web.bind.annotation.RequestMapping;
+        import org.springframework.web.bind.annotation.RestController;
+
+        import java.util.ArrayList;
+        import java.util.HashMap;
+        import java.util.Optional;
+
+        import static java.time.Year.isLeap;
+
+@RestController
+@RequestMapping("instructors")
+public class InstructorController {
+    @Autowired
+    InstructorService service;
```

Comment:
same changes as mentioned in student controller 
@RestController
@RequestMapping("instructors")
public class InstructorController {
    private final InstructorService service;
    private final DateValidationService dateValidationService;

    public InstructorController(InstructorService service, DateValidationService dateValidationService) {
        this.service = service;
        this.dateValidationService = dateValidationService;
    }

    @GetMapping("/time_table/{id}/{date}") //2.8.2023
    public ResponseEntity<String> getInstructorDayTimeTable(@PathVariable int id, @PathVariable String date){
        //check date is correct or not
        String[] date_array = date.split("-",3);

        int day = Integer.parseInt(date_array[0]);
        int month = Integer.parseInt(date_array[1]);
        int year = Integer.parseInt(date_array[2]);

        dateValidationService.validateDate(day, month, year);

        // Rest of your code...